### DISCUSSION TITLE
ListView with ItemTemplate and selecting DataTemplate from DataType

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#12793] [Q&A] [answered]
I've read in a lot of places (UWP, WinUI, etc) that you should be able to have multiple DataTemplates where the correct one is picked based on the DataType attribute. I have in vain tried to get this to work with a ListView. If I try to add more than one DataTemplate then it just says I can only have one child under the ListView.ItemTemplate node. I'm able to get it to work by using a DataTemplateSelector, but from what I understand it should be possible without that. Any suggestions/examples on how to get this to work?

**Comment by jeromelaban** [comment, marked as answer]
You may be referring to the [`HierarchicalDataTemplate` WPF feature](https://learn.microsoft.com/en-us/dotnet/api/system.windows.hierarchicaldatatemplate?view=windowsdesktop-7.0) that is not available in WinUI. `x:DataType` is only used in WinUI to specify the type of the DataContext [in the case of x:Bind](https://learn.microsoft.com/en-us/windows/uwp/data-binding/data-binding-quickstart#binding-to-a-collection-of-items).

The only way to implement template selection through types is by using `DataTemplateSelector`. The [TreeView control uses this](https://learn.microsoft.com/en-us/windows/apps/design/controls/tree-view#item-template-selectors) to represent items appropriately.

**Reply by HakanL** [reply]
Got it, sometimes it's hard to distinguish limitations in UWP vs WinUI vs Uno. I did end up with a DataTemplateSelector, which works, but it didn't feel like it was the best practice, but sounds like it may be.
